interfaces/ros1/wavemap_ros/src/map_operations/publish_pointcloud_operation.cc

```diff
@@ -0,0 +1,132 @@
+#include "wavemap_ros/map_operations/publish_pointcloud_operation.h"
+
+#include <sensor_msgs/PointCloud.h>
+#include <sensor_msgs/PointCloud2.h>
+#include <sensor_msgs/point_cloud_conversion.h>
+#include <wavemap/core/indexing/index_conversions.h>
+#include <wavemap/core/map/hashed_chunked_wavelet_octree.h>
+#include <wavemap/core/map/hashed_wavelet_octree.h>
+#include <wavemap/core/utils/iterate/grid_iterator.h>
+#include <wavemap/core/utils/profiler_interface.h>
+#include <wavemap_ros_conversions/geometry_msg_conversions.h>
+
+namespace wavemap {
+DECLARE_CONFIG_MEMBERS(PublishPointcloudOperationConfig,
+                      (once_every)
+                      (occupancy_threshold_log_odds)
+                      (only_publish_changed_blocks)
+                      (topic));
+
+bool PublishPointcloudOperationConfig::isValid(bool verbose) const {
+  bool all_valid = true;
+
+  all_valid &= IS_PARAM_GT(once_every, 0.f, verbose);
+  all_valid &= IS_PARAM_NE(topic, std::string(), verbose);
```

Comment:
No checks required on the other two options?

Reply:
For `only_publish_changed_blocks` both `true` and `false` are valid. The `occupancy_threshold_log_odds` is theoretically allowed to be any value between `-inf` and `inf`. But I'll add a check to forbid it from being `NaN`.
It could be nice to also check if it's within the `min_log_odds` and `max_log_odds` of the map, but we don't have access to the map when validating the config.